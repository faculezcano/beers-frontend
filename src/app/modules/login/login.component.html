<div class="container">
  <div class="row">
    <div class="col-md-4 mx-auto">
      <h1 class="text-muted">Login</h1>

      <div class="card">
        <div class="card-body">
          <form (submit)="$event.preventDefault()">
            <div class="mb-3">
              <label class="form-label" for="email">Email</label>
              <input
                #email
                class="form-control"
                placeholder="admin@example.com"
                type="email"
              />
            </div>

            <div class="mb-3">
              <label class="form-label" for="password">Password</label>
              <input
                #password
                [class]="'form-control' + (error ? ' is-invalid' : '')"
                placeholder="admin"
                type="password"
              />
              <div class="invalid-feedback">Invalid email or password</div>
            </div>

            <div class="text-end">
              <button
                class="btn btn-primary"
                (click)="loginClick(email.value, password.value)"
                [disabled]="loading"
              >
                <span
                  *ngIf="loading"
                  class="spinner-border spinner-border-sm"
                  role="status"
                  aria-hidden="true"
                ></span>
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
