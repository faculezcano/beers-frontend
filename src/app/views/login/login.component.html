<mat-form-field>
  <input #email matInput placeholder="Email" type="email" />
</mat-form-field>

<mat-form-field>
  <input #password matInput placeholder="Password" type="password" />
</mat-form-field>

<button mat-button (click)="loginClick(email.value, password.value)">
  Login
</button>
