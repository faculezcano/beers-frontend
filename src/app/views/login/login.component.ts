import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
import {AuthService} from "../../services/auth.service";

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss']
})
export class LoginComponent implements OnInit {

  public loading = false;

  constructor(
    private authService: AuthService
  ) { }

  ngOnInit(): void {
  }

  loginClick(email: string, password: string) {
    this.loading = true;
    this.authService.login(email, password);
  }

}
