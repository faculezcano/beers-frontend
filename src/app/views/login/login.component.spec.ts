import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginComponent } from './login.component';
import {HttpClientTestingModule} from "@angular/common/http/testing";
import {MatFormFieldModule} from "@angular/material/form-field";
import {MatInputModule} from "@angular/material/input";
import {MatButtonModule} from "@angular/material/button";
import {BrowserAnimationsModule} from "@angular/platform-browser/animations";

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        LoginComponent
      ],
      imports: [
        HttpClientTestingModule,
        MatFormFieldModule,
        MatInputModule,
        MatButtonModule,
        BrowserAnimationsModule
      ],
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  })
});
