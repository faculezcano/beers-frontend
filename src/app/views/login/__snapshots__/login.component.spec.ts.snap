// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match snapshot 1`] = `
<app-login
  __ngContext__={[Function LRootView]}
  authService={[Function AuthService]}
  loading="false"
>
  <mat-form-field
    class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
            data-placeholder="Email"
            matinput=""
            type="email"
          />
          <span
            class="mat-form-field-label-wrapper"
          >
            <label
              class="mat-form-field-label mat-empty mat-form-field-empty"
            >
              <span>
                Email
              </span>
            </label>
          </span>
        </div>
      </div>
      <div
        class="mat-form-field-underline"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
      >
        <div
          class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field><mat-form-field
    class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
            data-placeholder="Password"
            matinput=""
            type="password"
          />
          <span
            class="mat-form-field-label-wrapper"
          >
            <label
              class="mat-form-field-label mat-empty mat-form-field-empty"
            >
              <span>
                Password
              </span>
            </label>
          </span>
        </div>
      </div>
      <div
        class="mat-form-field-underline"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
      >
        <div
          class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field><button
    class="mat-focus-indicator mat-button mat-button-base"
    mat-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      Login
    </span>
    <span
      class="mat-ripple mat-button-ripple"
      matripple=""
    />
    <span
      class="mat-button-focus-overlay"
    />
  </button>
</app-login>
`;
